### PR TITLE
fix race condition in err goroutines; fix send on wrong goroutine

### PR DESCRIPTION
**Pull request by pdbogen**
Howdy!

One bug is pretty serious, which is that on line 172 of svg.go you attempt a send on the wrong error channel; the send should occur on `s.errors` instead of `g.errors`.

Someone less important but still worth fixing, in my opinion, is that the treatment of goroutines and channels with regard to errors has a race condition. Because the order of execution of goroutines is undefined without synchronizing events, it's possible for the `defer close(g.errors)` / `defer close(s.errors)` to actually execute before (or concurrently) with the sub-goroutine send on that channel; this results in a panic, because a send on a closed channel is a fatal error.

I fix this in the PR below with `sync.WaitGroup`, by ensuring that any sending goroutines have exited before closing the corresponding channel.

Thanks,
- Patrick

p.s., I also implemented handling of `h`/`H` and `v`/`V` as well as tests for these and more comprehensive and DRY tests for path parsing.

### CODE REVIEW

File: path.go
Comment (Shugyousha) [inline path.go]: Nitpick: I would prefer to have these two variable declarations before the function call.

File: path_test.go
Comment (Shugyousha) [inline path_test.go]: We should check the errors here. The tests will fail without checking (because the output will most likely be wrong) but in order to see what actually went wrong, iterating through the error channel with a require.NoError() is better.

File: path_test.go
Comment (Shugyousha) [inline path_test.go]: Let's keep the tests as quiet as possible and use t.Logf here instead of log.Printf.

Comment (Shugyousha) [inline path_test.go]: The tests can be simplified by just itereating over the DrawingInstructions like this:

```
+       for _, test := range tests {
+               svg, err := ParseSvg(test.Svg, "test", 0)
+               require.NoError(t, err)
+
+               dis, _ := svg.ParseDrawingInstructions()
+               strux := []*DrawingInstruction{}
+               for di := range dis {
+                       strux = append(strux, di)
+                       log.Printf("di: %+v, di.M: %+v", di, di.M)
+               }
+
+               if len(strux) != len(test.Kinds) {
+                       t.Fatalf("expected %d instructions for test %s, but received %d", len(test.Kinds), test.Description, len(strux))
+               }
+
+               for i, stru := range strux {
+                       if stru.Kind != test.Kinds[i] {
+                               t.Fatalf("expected instruction %d for test %s to be %d, but was %d", i, test.Description, test.Kinds[i], stru.Kind)
+                       }
+
+                       if stru.M == nil {
+                               continue
+                       }
+
+                       if stru.M[0] != test.XCoords[i] {
+                               t.Fatalf("expected X coordinate %d for test %s to be %f, but was %f", i, test.Description, test.XCoords[i], stru.M[0])
+                       }
+
+                       if stru.M[1] != test.YCoords[i] {
+                               t.Fatalf("expected Y coordinate %d for test %s to be %f, but was %f", i, test.Description, test.YCoords[i], stru.M[1])
+                       }
+               }
+       }
```

**Review by Shugyousha — CHANGES_REQUESTED** [review]
If you can change the t.Logf and the testing I will merge your patches after having tested them more thoroughly. Thanks for the patches!

**Comment by Shugyousha** [comment]
Ah, you should run `go fmt` over your code as the final step.

**Comment by pdbogen** [comment]
Thanks for taking a look, @Shugyousha. I've made the requested changes; please let me know if you find anything else that could use some TLC.

**Comment by Shugyousha** [comment]
Looks good! I have merged your commits into our dev branch. We will test them there and if we don't don't find any issues we will merge them into master. I will close this PR as soon as we have merged your patches into our master branch.

Thanks again!

**Review by Shugyousha — APPROVED** [review]

**Comment by Shugyousha** [comment]
Your commits are now in master, thanks!

**Pull request merged**
